### PR TITLE
add cal::relative_duration

### PR DESCRIPTION
**Pull request by fmoor**

### CODE REVIEW

File: tests/test_edgeql_expressions.py [hunk 1]
```diff
@@ -1357,8 +1357,8 @@ async def test_edgeql_expr_valid_order_06(self):
                 SELECT A ORDER BY A;
             ''',
             [
-                "19:01:22.306916",
-                "20:01:22.306916",
+                "PT19H1M22.306916S",
```
Comment (elprans) [inline tests/test_edgeql_expressions.py:1360]: Please add `cal::relative_delta` to the `VALUES` `dict` here to cover the standard operators you've added.

Comment (elprans) [reply tests/test_edgeql_expressions.py:1360]: By "here" I mean this file.

Comment (fmoor) [reply tests/test_edgeql_expressions.py:1365]: added

File: edb/graphql/types.py [hunk 2]
```diff
@@ -248,6 +248,7 @@ def parse_decimal_literal(
     'cal::local_datetime': GraphQLString,
     'cal::local_date': GraphQLString,
     'cal::local_time': GraphQLString,
+    'cal::reltaivedelta': GraphQLString,
```
Comment (elprans) [inline edb/graphql/types.py:251]: Missed a spot.

Comment (fmoor) [reply edb/graphql/types.py]: changed

**Review by elprans — COMMENTED** [review]
Looks good overall.  Please adjust the commit title and add a more descriptive commit message.  Also, please document the new type.

**Comment by fmoor** [comment]
I made an attempt at documentation. It probably needs tweaking.

File: docs/datamodel/scalars/datetime.rst [hunk 3]
```diff
@@ -182,6 +182,66 @@ EdgeDB stores and outputs timezone-aware values in UTC.
...
+    - ``'decades'``
+    - ``'centuries'``
+    - ``'millennium'``
+
+    .. code-block:: edgeql
+
+        SELECT <duration>'45.6 seconds';
+        SELECT <duration>'15 milliseconds';
+        SELECT <duration>'3 weeks 45 minutes';
+        SELECT <duration>'-7 millennium';
```
Comment (elprans) [inline docs/datamodel/scalars/datetime.rst:227]:
```suggestion
        SELECT <cal::relative_duration>'45.6 seconds';
        SELECT <cal::relative_duration>'15 milliseconds';
        SELECT <cal::relative_duration>'3 weeks 45 minutes';
        SELECT <cal::relative_duration>'-7 millennium';
```

**Review by elprans — APPROVED** [review]

File: docs/datamodel/scalars/datetime.rst [hunk 4]
```diff
@@ -182,6 +182,66 @@ EdgeDB stores and outputs timezone-aware values in UTC.
...
 
 
+----------
+
+
+.. eql:type:: cal::relative_duration
+
+    A type representing a span of time.
+
+    Unlike :eql:type:`std::duration` a relative_duration is not an precise
```
Comment (1st1) [inline docs/datamodel/scalars/datetime.rst:192]:
```suggestion
    Unlike :eql:type:`std::duration` a relative_duration is not a precise
```

File: docs/datamodel/scalars/datetime.rst [hunk 5]
```diff
@@ -182,6 +182,66 @@ EdgeDB stores and outputs timezone-aware values in UTC.
...
+        db> select (
+        ...     <datetime>'2020-01-01T00:00:00Z' +
+        ...     <cal::relative_duration>'1 year'
+        ... ) - <datetime>'2020-01-01T00:00:00Z';
+        {<duration>'8784:00:00'}
+        db> select (
+        ...     <datetime>'2019-01-01T00:00:00Z' +
+        ...     <cal::relative_duration>'1 year'
+        ... ) - <datetime>'2019-01-01T00:00:00Z';
+        {<duration>'8760:00:00'}
```
Comment (1st1) [inline docs/datamodel/scalars/datetime.rst:206]: Hm, this example is very hard to parse -- please add a few comments explaining what happened in the first query, what happened in the last, why the first result is that, and what's up with the second result.

File: docs/datamodel/scalars/datetime.rst [hunk 6]
```diff
@@ -182,6 +182,66 @@ EdgeDB stores and outputs timezone-aware values in UTC.
...
 
+----------
+
+
+.. eql:type:: cal::relative_duration
+
+    A type representing a span of time.
+
+    Unlike :eql:type:`std::duration` a relative_duration is not an precise
+    measurment. For example:
```
Comment (1st1) [inline docs/datamodel/scalars/datetime.rst:193]: Please expand on what "not a precise measurement" actually means.

File: docs/edgeql/funcops/datetime.rst [hunk 7]
```diff
@@ -67,10 +70,22 @@ Date and Time
 
 
 .. eql:operator:: DTPLUS: datetime + duration -> datetime
+                          datetime + cal::relative_duration \
```
Comment (1st1) [inline docs/edgeql/funcops/datetime.rst:73]: I'd keep the `duration` group as is, and add the `relative_durations` ops after it. I.e. don't interleave them like that, it will be hard to read/follow.

File: docs/edgeql/funcops/datetime.rst [hunk 8]
```diff
@@ -90,16 +105,27 @@ Date and Time
...
+                                -> cal::relative_duration
+                           cal::relative_duration - cal::relative_duration \
+                                -> cal::relative_duration
                            datetime - datetime -> duration
                            cal::local_datetime - cal::local_datetime \
                                 -> duration
                            local_time - local_time -> duration
                            local_date - local_date -> duration
                            datetime - duration -> datetime
+                           datetime - cal::relative_duration -> datetime
```
Comment (1st1) [inline docs/edgeql/funcops/datetime.rst:120]: ditto

**Review by 1st1 — APPROVED** [review]
Great job; please fix the nit comments I left before merging.

File: docs/datamodel/scalars/datetime.rst (hunk 5 above)
Comment (fmoor) [reply docs/datamodel/scalars/datetime.rst:212]: I think I improved this. Not sure it's optimal though.
